### PR TITLE
test(windows): make Windows kill child processes when the parent exits

### PR DESCRIPTION
**Pull request by ReactorScram**
Probably needs some refactoring and prettifying before it comes out of drafts.

### CODE REVIEW

**Comment by ReactorScram** [comment]
It's so much code. I thought IPC was gonna be simple 😬

File: rust/windows-client/src-tauri/src/client/deep_link.rs [hunk 1]
```diff
@@ -118,6 +118,7 @@ impl Server {
         }
 
         let mut sa = WinSec::SECURITY_ATTRIBUTES {
+            // TODO: Try `size_of_val` here instead
```
Comment (conectado) [inline rust/windows-client/src-tauri/src/client/deep_link.rs:121]: is this outdated?

File: rust/windows-client/src-tauri/src/client/ipc.rs [hunk 2]
```diff
@@ -271,6 +402,45 @@ impl Drop for SubcommandChild {
...
+    job_object: HANDLE,
+}
+
+impl LeakGuard {
+    pub fn new() -> Result<Self> {
+        let job_object = unsafe { CreateJobObjectA(None, None) }?;
+
+        let mut jeli = JOBOBJECT_EXTENDED_LIMIT_INFORMATION::default();
+        jeli.BasicLimitInformation.LimitFlags = JOB_OBJECT_LIMIT_KILL_ON_JOB_CLOSE;
+        // SAFETY: Windows shouldn't hang on to `jeli`. I'm not sure why this is unsafe.
```
Comment (conectado) [inline rust/windows-client/src-tauri/src/client/ipc.rs:416]: All FFI is unsafe by default

**Review by conectado — APPROVED** [review]

File: rust/windows-client/src-tauri/src/client/ipc.rs (hunk 2 above)
Comment (ReactorScram) [reply rust/windows-client/src-tauri/src/client/ipc.rs:416]: Yeah but the `windows` crate is supposed to have safe wrappers around some of the unsafe functions from `windows-sys`. Like here the XML document is safe https://github.com/microsoft/windows-rs/blob/master/crates/samples/windows/xml/src/main.rs

I guess they are making those safe wrappers by hand or something. For many of the new APIs I used the last week or two, they're only offering the auto-generated unsafe functions that take pointers.

File: rust/windows-client/src-tauri/src/client/deep_link.rs (hunk 1 above)
Comment (ReactorScram) [reply rust/windows-client/src-tauri/src/client/deep_link.rs:121]: No, I just didn't want to risk changing the deep link module in this PR

**Pull request merged**